File: 0003_Time_Conversion.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'timeConversion' function below.
#
# The function is expected to return a STRING.
# The function accepts STRING s as parameter.
#

def timeConversion(s):
    # Write your code here
    
    time_components = s.split(":")
    
    hour = time_components[0]
    minute = time_components[1]
    seconds = time_components[2]
    
    if "PM" in seconds:
        seconds = seconds.replace("PM", "")
        if hour != "12":
            hour = int(hour) + 12

    elif "AM" in seconds:
        seconds = seconds.replace("AM", "")
        if hour == "12":
            hour = "00"
    
    military_time_format = f"{hour}:{minute}:{seconds}"
    
    return military_time_format

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    s = input()

    result = timeConversion(s)

    fptr.write(result + '\n')

    fptr.close()

File: 0003_Time_Conversion.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Result
{

    /*
     * Complete the 'timeConversion' function below.
     *
     * The function is expected to return a STRING.
     * The function accepts STRING s as parameter.
     */

    public static string timeConversion(string s)
    {
        string[] words = s.Split(':');
        
        byte hour = Byte.Parse(words[0]);
        byte minute = Byte.Parse(words[1]);
        byte second = Byte.Parse(words[2].Substring(0, 2));
        string period = words[2].Substring(2, 2);
        
        if (period == "PM" & hour != 12)
        {
            hour += 12;
        }
        else if (period == "AM" & hour == 12)
        {
            hour = 0;
        }
        
        string FormattedMilitaryTime = $"{hour.ToString("00")}:{minute.ToString("00")}:{second.ToString("00")}";
        
        Console.WriteLine(FormattedMilitaryTime);
        return FormattedMilitaryTime;
    }

}

class Solution
{
    public static void Main(string[] args)
    {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        string s = Console.ReadLine();

        string result = Result.timeConversion(s);

        textWriter.WriteLine(result);

        textWriter.Flush();
        textWriter.Close();
    }
}
